### ISSUE TITLE
How to add .html file in sweetalert

### ISSUE THREAD

**Issue by pritesh1209** [sweetalert2/sweetalert2-examples#4]
Hi,
I would like to inject .html file in swal or angular component.
for example,
```javascript
swal({
  title: 'hello',
  buttonsStyling: false,
  confirmButtonClass: 'btn btn-info',
  html: // index.html or AddCustomerComponent
}).catch(swal.noop);
```
????

**Comment by gverni** [comment]
Hi @pritesh1209 the `html` property of swal gets a string, an HTML element or a JQuery object containing the html code. If you want to inject an external html file, then you need to load it first (e.g. through AJAX or `fetch` call) and then assign it to the `html` property. You may want to use the `onBeforeOpen` property to define a function that fetches the html file before the modal is open.

**Comment by limonte** [comment]
@gverni's answer is correct, for any further assistance please use StackOverflow: https://stackoverflow.com/questions/ask

**Comment by limonte** [comment]
@gverni what are your thoughts on answering `howto` questions in this repo? IMO it's better to forward people to StackOverflow so they will get the answers more possibly, what are your thoughts?

**Comment by pritesh1209** [comment]
Can you please give me one example?

**Comment by gverni** [comment]
> what are your thoughts on answering howto questions in this repo?

@limonte I think there are pros and cons in using StackOverflow for how to questions. Let me try to summarize my view about StackOverflow: 

* Pros
  * We leverage the help of the community, since it's more likely developers are _watching_ topics on StackOverflow than the sweetalert repo
  * For issue related to integration with other framework (React, vue, Angular, etc) you have more experts providing answers
  * StackOverflow results are always top of the list on google search 
  * You have less obligation to promptly respond to StackOverflow questions (this may be debatable though, since it's still _our_ product)
* Cons
  * StackOverflow answers are not moderated / have no ownership. I've seen answers of StackOverflow that weren't the best way of achieving something. I know you can edit and _downvote_ answers, but that means we have to _police_ StackOverflow as well.   
  * Follow-ups and conversations are practically impossible on StackOverflow. It only accepts answers and comments. GitHub on the other hands allows for conversations to flow much better and to link pieces of code / issues / PRs / releases, etc
  * Notification on StackOverflow is not as effective as GitHub with mentions (or I haven't figured out a way of making it work)
  * Some howto questions are actually source for interesting new advanced example (or sometimes issues which are not immediately clear). Again, unless we are _policing_ StackOverflow (which we can only do when we are not working on improving SweetAlert) we may miss them. That's the reason why I want to transfer How-To questions to this repo. 
  * It's much easier to create an issue from an how to question on GitHub than from StackOverflow 

**TLDR** 
I may be biased because I think GitHub is a very effective and powerful platform to keep everything in one place (instead of going around to check other websites). It makes the software development workflow more effective and our life easier. We maybe need to just accept that we may not be as quick to answer how-to questions. 

From the user standpoint, receiving an answer from us (or from the community) directly on the GitHub repo gives that blessing that they are looking for (since it comes or has been approved directly from the developers of Sweetalert2)

So, I'm keen to answer how-to questions here and redirect to StackOverflow if the answer is beyond our reach (e.g. is about integration with other framework we don't know about)

What do you think @limonte?

**Comment by gverni** [comment]
@pritesh1209 you can find at [this url](https://swal2-fetch-html.glitch.me/) a possible way of implementing it. I'm not an expert with Angular, so you will have to integrate that yourself in your project. But since the example is using just one API (i.e. `fetch`) it should be fairly easy. 

Note that in this example I'm using a different (simpler) approach than using the `onBeforeOpen`.

**Comment by limonte** [comment]
I agree on all pros and cons you mentioned @gverni. For me personally, it makes more sense to separate concerns ([SoC](https://en.wikipedia.org/wiki/Separation_of_concerns)) and have issue-tracker and the place for questions separated.

I see GitHub issues as the backlog and obviously, there's no place for questions there.  That's why the "question" issue template just forwards users to SO: https://github.com/sweetalert2/sweetalert2/issues/new?template=03_question.md

There's one SO pros you forgot - the culture of asking questions. SO has all the tools to remind users to ask questions **properly**. Poorly formulated questions will be downvoted of simply left w/o any answers. GH on the other hand doesn't have that and we need to manually remind to provide more details.

For instance, this question. It has absolutely nothing to do with SweetAlert2, for me it looks like "I'm using heavily outdated version of SweetlAlert2 (probably v6) and my favorite cool framework, do my homework for me please." You might call me unpolite, but the best I can do for these guys is to answer with https://stackoverflow.com/questions/ask and close the ticket :)

But, you're the boss on this repo, so feel free to go with whatever you feel like.

**Comment by gverni** [comment]
> the best I can do for these guys is to answer with https://stackoverflow.com/questions/ask and close the ticket :)

I do understand you, and it gets really frustrating at times. On the other hand, sometime there are people that are genuinely struggling with what looks to us simple concepts. I know because I was one of them not long ago :smile:

> But, you're the boss on this repo, so feel free to go with whatever you feel like.

You know that your opinions are very important to me @limonte and thanks for sharing it. But I want to give it a try. Maybe later I will regret it and I'll start to direct people back to SO 😉 I'll try to transfer all the howto question from the main repo to this. Feel free to do the same if you want.
